test/functional/mining_basic.py

```diff
@@ -164,9 +164,9 @@ def assert_submitblock(block, result_str_1, result_str_2=None):
         assert_submitblock(bad_block, 'prev-blk-not-found', 'prev-blk-not-found')
 
         self.log.info('submitheader tests')
-        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='xx' * 80))
-        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='ff' * 78))
-        assert_raises_rpc_error(-25, 'Must submit previous header', lambda: node.submitheader(hexdata='ff' * 80))
+        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='xx' * BLOCK_HEADER_SIZE))
+        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='ff' * (BLOCK_HEADER_SIZE-2)))
+        assert_raises_rpc_error(-25, 'Must submit previous header', lambda: node.submitheader(hexdata=super(CBlock, bad_block).serialize().hex()))
```

Comment:
I'm not sure I follow here: which constructor is it that takes `CBlock` and `bad_block` as parameters?

Reply:
The base class is CBlockHeader, so we're accessing its serialization function.

Reply:
Maybe rename `bad_block` above to `bad_hashPrevBlock_block` or similar, and invoke the `CBlockHeader` constructor directly?